External/Plugins/CodeRefactor/Commands/Rename.cs

```diff
@@ -5,119 +5,160 @@
     /// </summary>
-    public class Rename : RefactorCommand<IDictionary<String, List<SearchMatch>>>
+    public class Rename : RefactorCommand<IDictionary<string, List<SearchMatch>>>
     {
-        private Boolean outputResults;
+        private static bool includeComments = false;
+        private static bool includeStrings = false;
+        private static bool previewChanges = true;
+
+        private bool isRenamePackage;
+        private string renamePackagePath;
         private FindAllReferences findAllReferencesCommand;
         private Move renamePackage;
-        private String oldFileName;
-        private String newFileName;
+        private IRenameHelper helper;
 
-        public string TargetName { get; private set; }
+        private string oldFileName;
+        private string newFileName;
+
+        public string OldName { get; private set; }
         public string NewName { get; private set; }
+        public bool OutputResults { get; private set; }
+        public ASResult Target { get; private set; }
+        public string TargetName { get; private set; }
 
         /// <summary>
         /// A new Rename refactoring command.
+        /// Doesn't use inline.
         /// Outputs found results.
         /// Uses the current text location as the declaration target.
         /// </summary>
-        public Rename()
-            : this(true)
+        public static Rename Create()
         {
+            return Create(false);
         }
 
         /// <summary>
         /// A new Rename refactoring command.
+        /// Outputs found results.
         /// Uses the current text location as the declaration target.
         /// </summary>
+        /// <param name="inline">Whether to use inline renaming.</param>
+        public static Rename Create(bool inline)
+        {
+            return Create(inline, true);
+        }
+
+        /// <summary>
+        /// A new Rename refactoring command.
+        /// Uses the current text location as the declaration target.
+        /// </summary>
+        /// <param name="inline">Whether to use inline renaming.</param>
         /// <param name="outputResults">If true, will send the found results to the trace log and results panel</param>
-        public Rename(Boolean outputResults)
-            : this(RefactoringHelper.GetDefaultRefactorTarget(), outputResults)
+        public static Rename Create(bool inline, bool outputResults)
         {
+            return Create(RefactoringHelper.GetDefaultRefactorTarget(), inline, outputResults);
         }
 
         /// <summary>
         /// A new Rename refactoring command.
         /// </summary>
+        /// <param name="inline">Whether to use inline renaming.</param>
         /// <param name="target">The target declaration to find references to.</param>
         /// <param name="outputResults">If true, will send the found results to the trace log and results panel</param>
-        public Rename(ASResult target, Boolean outputResults)
```

Comment:
I think it's a bad idea, because all other commands look different. and I do not understand why it was necessary to change the constructors may be back as it was?